### PR TITLE
Add tutorial for sensitive

### PR DESCRIPTION
**Pull request by wanghaoshuang**
Add tutorial for sensitive.

### CODE REVIEW

File: slim/sensitive/sensitive.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    for key, value in kwargs.items():
+        if os.environ.get(key, None) is None:
+            os.environ[key] = str(value)
+
+
+# NOTE(paddle-dev): All of these flags should be set before
+# `import paddle`. Otherwise, it would not take any effect.
+set_paddle_flags(
+    FLAGS_eager_delete_tensor_gb=0,  # enable GC to save memory
+)
```
Comment (qingqing01) [inline slim/sensitive/sensitive.py]: Remove line 26 to line 36, these also will be removed in tools/train.py tools/eval.py tools/infer.py.

File: slim/sensitive/sensitive.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+def main():
+    env = os.environ
+
+    cfg = load_config(FLAGS.config)
+    assert 'architecture' in cfg
+    main_arch = cfg.architecture
+
+    merge_config(FLAGS.opt)
+
+    print_total_cfg(cfg)
```
Comment (qingqing01) [inline slim/sensitive/sensitive.py]: Maybe remove `print_total_cfg` in this file

File: slim/sensitive/sensitive.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    devices_num = fluid.core.get_cuda_device_count()
```
Comment (qingqing01) [inline slim/sensitive/sensitive.py]: devices_num is not used

File: slim/sensitive/sensitive.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+    train_feed = create(cfg.train_feed)
```
Comment (qingqing01) [inline slim/sensitive/sensitive.py]: train_feed is not used

File: slim/sensitive/sensitive.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+    # build program
+    startup_prog = fluid.Program()
+    eval_prog = fluid.Program()
+    with fluid.program_guard(eval_prog, startup_prog):
+        with fluid.unique_name.guard():
+            model = create(main_arch)
+            eval_loader, feed_vars = create_feed(eval_feed)
+            fetches = model.eval(feed_vars)
+    eval_prog = eval_prog.clone(True)
```
Comment (qingqing01) [inline slim/sensitive/sensitive.py]: master分支reader构建以改变，需参考 tools/eval.py下面代码

File: slim/sensitive/sensitive.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+    ignore_params = cfg.finetune_exclude_pretrained_params \
+                 if 'finetune_exclude_pretrained_params' in cfg else []
+
+    start_iter = 0
+    if cfg.pretrain_weights and fuse_bn and not ignore_params:
+        checkpoint.load_and_fusebn(exe, eval_prog, cfg.pretrain_weights)
+    elif cfg.pretrain_weights:
+        checkpoint.load_params(
+            exe, eval_prog, cfg.pretrain_weights, ignore_params=ignore_params)
```
Comment (qingqing01) [inline slim/sensitive/sensitive.py]: 敏感度评估加载的是weights， 而不是pretrain_weights吧

File: slim/sensitive/sensitive.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        "-r",
+        "--pruned_ratios",
+        default="0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9",
+        type=str,
+        help="The ratios pruned iteratively for each parameter when calculating sensitivities.")
+    parser.add_argument(
+        "-P",
+        "--print_params",
+        default=False,
+        type=bool,
```
Comment (heavengate) [inline slim/sensitive/sensitive.py]: 可以用action类型，类似tools/train.py里面的--eval

**Comment by wanghaoshuang** [comment]
让用户显式的指定要裁剪的参数更保守安全些，避免一些参数被误裁剪。

**Pull request merged**